modules/nf-scil/registration/tractogram/main.nf

```diff
@@ -0,0 +1,86 @@
+process REGISTRATION_TRACTOGRAM {
+    tag "$meta.id"
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'scil.usherbrooke.ca/containers/scilus_1.6.0.sif':
+        'scilus/scilus:1.6.0' }"
+
+    input:
+    tuple val(meta), path(anat), path(transfo), path(tractograms_dir, stageAs: 'tractograms/'), path(ref) /* optional, value = [] */, path(deformation) /* optional, value = [] */
+
+    output:
+    tuple val(meta), path("*__*.{trk,tck}"), emit: warped_tractogram
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def reference = "$ref" ? "--reference $ref" : ""
+    def in_deformation = "$deformation" ? "--in_deformation $deformation" : ""
+
+    def inverse = task.ext.inverse ? "--inverse" : ""
+    def reverse_operation = task.ext.reverse_operation ? "--reverse_operation" : ""
+    def invalid_action = task.ext.invalid_action ? "--" + task.ext.invalid_action : "--remove_invalid"
+    def no_empty1 = task.ext.no_empty1 ? "--no_empty" : ""
+    def force = task.ext.force ? "-f" : ""
+
+    def cut_invalid = task.ext.cut_invalid ? "--cut_invalid" : ""
+    def remove_single_point = task.ext.remove_single_point ? "--remove_single_point" : ""
+    def remove_overlapping_points = task.ext.remove_overlapping_points ? "--remove_overlapping_points" : ""
+    def threshold = task.ext.threshold ? "--threshold " + task.ext.threshold : ""
+    def no_empty2 = task.ext.no_empty2 ? "--no_empty" : ""
+
+    """
+    for tractogram in ${tractograms_dir};
+        do bname=\${tractogram/_centroid/}
```

Comment:
This way of extracting the output name from the tractogram is a bit outdated. I'm not sure it'll continue to apply consistently. A better way would be to use the tractogram name in full as output name. Since all input tractograms are placed in a directory, there is no overwritting problem : `bname=\${tractogram}`

However, to remain consistent, the output names do have to start by the `prefix` defined in the script. A tractogram could start with `prefix`, so you have to remove it if it is there : `bname=\${tractogram/${prefix}_/_}` (if it's not, the command is still valid, no need to put a condition around to validate).